.github/workflows/ci.yml

```diff
@@ -305,4 +305,22 @@ jobs:
           then
             echo "PR title is too long (greater than 72 characters)"
             exit 1
-          fi
\ No newline at end of file
+          fi
+
+  check-proto-files:
+    name: Check for changes in proto files
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions-rs/toolchain@16499b5e05bf2e26879000db0c1d13f7e13fa3af # v1.0.7
+        with:
+          profile: minimal
+          toolchain: stable
+          override: true
+
+      - name: Generated proto files and diff
+        run: |
+          cargo install pb-rs 
```

Comment:
Two things: Use `--locked` and if you add the cache action before this, we don't have to reinstall it every time.